android/src/org/kiwix/kiwixmobile/ZimFileSelectFragment.java

```diff
@@ -381,47 +415,4 @@ public View getView(int position, View convertView, ViewGroup parent) {
       ImageView favicon;
     }
   }
-
-  // This AsyncTask will scan the file system for files with the Extension ".zim" or ".zimaa"
-  private class RescanFileSystem extends AsyncTask<Void, Void, Void> {
-
-    @Override
-    protected void onPreExecute() {
-
-      super.onPreExecute();
-    }
-
-    @Override
-    protected Void doInBackground(Void... params) {
-      // Search file system for files
-      mFiles = new FileSearch().findFiles(context);
-
-      // Remove duplicate files
-      HashSet<LibraryNetworkEntity.Book> singularBooks = new HashSet<>();
-      singularBooks.addAll(mFiles);
-      mFiles.clear();
-      mFiles.addAll(singularBooks);
-
-      // Sort files
-      Collections.sort(mFiles, new fileComparator());
-
-      // Save files for quick access later
-      bookDao.saveBooks(mFiles);
```

Comment:
This here serves to refresh the database when files are deleted. If the database is not refreshed then the deleted files don't show as downloadable in the LibraryAdapter.java adapter for our library fragment. You need to re implement this somehow. (Note simply calling BookDao.getBooks() should be enough to check if all the files exist and remove where necessary).